packages/elements/src/photon-medication-search/photon-medication-search-component.tsx

```diff
@@ -0,0 +1,350 @@
+          {t.name ? <span class="text-blue-600">({boldSubstring(t.name, searchText)}): </span> : ''}
+          {boldSubstring(t.treatment.name, searchText)}
+        </div>
+        <div class="text-xs text-gray-500 truncate">
+          {t.dispenseQuantity} {t.dispenseUnit}, {t.daysSupply}{' '}
+          {t.daysSupply === 1 ? 'day' : 'days'} supply, {refills}{' '}
+          {refills === 1 ? 'refill' : 'refills'}, {t.instructions}
+        </div>
+      </div>
+    );
+  }
+
+  return t.treatment.name;
+}
+
+function getGroupsConfig(props: ComponentProps) {
+  return [
+    {
+      label: 'OFF CATALOG',
+      filter: (t: any) =>
+        t && typeof t === 'object' && props.offCatalogOption && t.id === props.offCatalogOption.id
+    },
+    {
+      label: 'PERSONAL TEMPLATES',
+      filter: (t: any) => t && typeof t === 'object' && 'treatment' in t && t.isPrivate
+    },
+    {
+      label: 'ORGANIZATION TEMPLATES',
+      filter: (t: any) => t && typeof t === 'object' && 'treatment' in t && !t.isPrivate
+    },
+    {
+      label: 'ORGANIZATION CATALOG',
+      filter: (t: any) =>
+        t && typeof t === 'object' && 'name' in t && !('treatment' in t) && !('medicationId' in t)
+    },
+    {
+      label: 'ALL TREATMENTS',
+      filter: (t: any) => t && typeof t === 'object' && 'medicationId' in t
+    }
+  ];
+}
+
+// Component Definition
+
+interface ComponentProps {
+  label?: string;
+  required: boolean;
+  invalid: boolean;
+  helpText?: string;
+  catalogId?: string;
+  disabled: boolean;
+  formName?: string;
+  selected?: Treatment | PrescriptionTemplate | TreatmentOption;
+  offCatalogOption?: Medication;
+}
+
+const Component = (props: ComponentProps) => {
+  let ref: any;
+  const client = usePhoton();
+  const { store, actions } = CatalogStore;
+  const [searchText, setSearchText] = createSignal<string>('');
+  const [treatmentOptions, setTreatmentOptions] = createSignal<TreatmentOption[]>([]);
+  const [enableFullWidthMedicationSearch, setEnableFullWidthMedicationSearch] =
+    createSignal<boolean>(false);
+
+  onMount(async () => {
+    await actions.getCatalogs(client!.getSDK());
+  });
+
+  createMemo(async () => {
+    const options = await loadTreatmentOptions(client!.sdk.apolloClinical, searchText());
+    setTreatmentOptions(options);
+  });
+
+  const data = createMemo(() => {
+    const filteredData = getFilteredData(props, searchText(), treatmentOptions());
+    return filteredData.length > 0 ? filteredData : [];
+  });
+
+  const displayAccessor = (
+    t: Treatment | PrescriptionTemplate | TreatmentOption,
+    groupAccess: boolean
+  ) => {
+    if (t && '__typename' in t && t.__typename == 'PrescriptionTemplate') {
+      return displayPrescriptionTemplate(t as PrescriptionTemplate, groupAccess, searchText());
+    }
+    return displayTreatment(t as Treatment | TreatmentOption, groupAccess, searchText());
+  };
+
+  const isMobile = window.innerWidth < 768;
```

Comment:
Is there no built in way to switch what gets rendered based on screen size? I'm worried this wont work if you resize the browser

Reply:
nope, not that i know of

Reply:
why would they resize?

Reply:
Rotation, a bunch of other cases

Reply:
It does exist but can punt for now I guess https://primitives.solidjs.community/package/media